bigframes/core/compile/compiler.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import dataclasses
+import functools
+import typing
+
+import pandas as pd
+
+import bigframes.core.compile as compiled
+import bigframes.core.compile.single_column
+import bigframes.core.nodes as nodes
+
+if typing.TYPE_CHECKING:
+    import bigframes.core
+    import bigframes.session
+
+
+@dataclasses.dataclass(frozen=True)
+class ArrayValueCompiler:
```

Comment:
This class isn't necessary.

When we need a hashable context (e.g. ordering), we can add that parameter when we need it.

Reply:
done